### ISSUE TITLE
Add vaults to squid

### ISSUE THREAD

**Issue by nud3l** [interlay/interbtc-squid#51]
## Goal

Add a `vaults` endpoint where we can the data available at https://app.interlay.io/dashboard/vaults

## Implementation

Vaults have:

- AccountId
- Identity
- Locked collateral (updated on registration, deposit and withdraw collateral)
- Pending BTC
- Collateralization (based on current collateral and exchange rate)
- Status issuing (enabled disabled)
- Status collateral (see https://docs.interlay.io/#/vault/overview?id=collateral-thresholds)

## Replaces

![image](https://user-images.githubusercontent.com/14966470/211491144-64b14a4c-4ae9-43d4-824e-51179015074c.png)

**Comment by nud3l** [comment]
Adding it to the existing vault object is the right approach. Thank you!

**Comment by jtfirek** [comment]
Sounds good!

**Comment by jtfirek** [comment]
@nud3l The locked collateral in the dashboard displays the collateral as DOT. Is the collateral in the multi-collateral system all converted to DOT equivalent value for displaying here?

**Comment by nud3l** [comment]
No, there are individual dashboards for each collateral asset. The DOT is just due to Interlay only having DOT as collateral right now. See Kintsugi as an example: https://kintsugi.interlay.io/dashboard/vaults

There's then a summary dashboard that converts all collateral assets to USD but that would be an add on and I'd consider it out of scope for this PR.

**Comment by jtfirek** [comment]
Ok thanks that clears up all my confusion.

**Comment by jtfirek** [comment]
Looks like the proposed goal does not included locked btc amount just the a pending btc amount. I feel that we need locked btc to calculation the collateratlization: 
`collateralization = colleratalAmount (converted to btc value) / locked btc`

**Comment by nud3l** [comment]
great catch, I forgot to add it.

**Comment by jtfirek** [comment]
@nud3l For the status issuing, Is this referring to 
`status | VaultStatus | Current status of the vault (Active, Liquidated, CommittedTheft)`
from the iBTC spec?

**Comment by nud3l** [comment]
Almost:

StatusIssuing - Active or Inactive (just that part basically: https://github.com/interlay/interbtc/blob/master/crates/vault-registry/src/types.rs#L132)

StatusCollateral - AboveSecureThreshold, BelowSecureThreshold, BelowPremiumRedeemThreshold, Liquidated. The parachain just tracks if a vault is liquidated but the UI/frontend would also track if a vault is above or below certain collateral thresholds as determined by the ones here: https://docs.interlay.io/#/vault/overview?id=collateral-thresholds
